File: temp/find_syntax_error.js
const fs = require('fs');

const code = fs.readFileSync("C:/Users/Love/Documents/Codex/TKBCherry/web/tkb-fet-engine.js", 'utf8');
const lines = code.split("\n");

console.log("=== CHECKING LINES AROUND anyImproved ===");
for (let i = 0; i < lines.length; i++) {
  if (lines[i].includes("anyImproved") || lines[i].includes("tryRelaxAndRepairGapGaps") || lines[i].includes("tryCrushExtremeSpanGaps")) {
    console.log(`Line ${i+1}: ${lines[i]}`);
  }
}

File: scratch/find_syntax_error.py
import sys, codecs
sys.stdout.reconfigure(encoding='utf-8')

engine_file = r'C:\Users\Love\Documents\Codex\TKBCherry\web\pages\tkb-fet-engine.js'

with codecs.open(engine_file, 'r', 'utf-8') as f:
    lines = f.readlines()

for i, l in enumerate(lines):
    if 'stageDeadlineMs' in l:
        print(f"Line {i+1}: {l}")

